### PR TITLE
[UCAI] Add anthropic integration to the unity catalog ai toolkit

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Adds support for the Anthropic SDK for tool usage with functions stored within unity catalog

### CODE REVIEW

File: unitycatalog-ai/integrations/anthropic/README.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+# Using Unity Catalog AI with the Anthropic SDK
+
+You can use the Unity Catalog AI package with the Anthropic SDK to utilize functions that are defined in Unity Catalog to be used as tools within Anthropic LLM calls.
+
+## Installation
+
+```sh
+# install from the source
+pip install git+https://github.com/serena-ruan/unitycatalog-ai.git#subdirectory=integrations/anthropic
```
Comment (serena-ruan) [inline unitycatalog-ai/integrations/anthropic/README.md:9]:
```suggestion
pip install git+https://github.com/unitycatalog/unitycatalog.git#subdirectory=unitycatalog-ai/integrations/anthropic
```

File: .github/workflows/ucai-integration-tests.yml [hunk 2]
```diff
@@ -36,3 +36,27 @@ jobs:
...
+            uses: actions/setup-python@v5
+            with:
+              python-version: ${{ matrix.python-version }}
+          - name: Install dependencies
+            run: |
+              pip install .
+              pip install integrations/anthropic[dev]
+          - name: Run tests
+            run: |
+              pytest integrations/anthropic/tests
```
Comment (serena-ruan) [inline .github/workflows/ucai-integration-tests.yml:62]: The format seems not correct
```suggestion
      runs-on: ubuntu-latest
      strategy:
        matrix:
          python-version: ['3.9', '3.10']
      timeout-minutes: 20
      defaults:
          run:
              working-directory: unitycatalog-ai
      steps:
        - name: Checkout code
           uses: actions/checkout@v4
        - name: Set up Python
           uses: actions/setup-python@v5
           with:
             python-version: ${{ matrix.python-version }}
        - name: Install dependencies
           run: |
             pip install .
             pip install integrations/anthropic[dev]
        - name: Run tests
           run: |
             pytest integrations/anthropic/tests
```

**Review by serena-ruan — APPROVED** [review]
LGTM! Let's address the comments before merge!

File: .github/workflows/ucai-integration-tests.yml (hunk 2 above)
Comment (BenWilson2) [reply .github/workflows/ucai-integration-tests.yml:62]: weird auto-tab applied in IDE?

**Comment by BenWilson2** [comment]
@dennyglee this one is all set!

**Review by dennyglee — APPROVED** [review]

**Comment by dennyglee** [comment]
Thanks @BenWilson2 and @serena-ruan  - LGTM!

**Pull request merged**